iceoryx_posh/include/iceoryx_posh/internal/popo/building_blocks/unique_port_id.hpp

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+#ifndef IOX_POSH_POPO_BUILDING_BLOCKS_TYPED_UNIQUE_ID_HPP
+#define IOX_POSH_POPO_BUILDING_BLOCKS_TYPED_UNIQUE_ID_HPP
+
+#include "iceoryx_hoofs/cxx/newtype.hpp"
+#include "iceoryx_hoofs/error_handling/error_handling.hpp"
+
+#include <atomic>
+#include <cstdint>
+#include <limits>
+
+namespace iox
+{
+namespace popo
+{
+/// @brief Struct to signal the constructor to create an invalid id
+struct InvalidPortId_t
+{
+};
+constexpr InvalidPortId_t InvalidPortId = InvalidPortId_t();
+
+/// @brief Unique ID depending on a type. If you would like to assign different
```

Comment:
Could you elaborate the usage of the RouDi ID a bit further here?

Reply:
I try ... please close if you are satisfied